librustzcash/src/rustzcash.rs

```diff
@@ -399,15 +399,11 @@ fn test_gen_r() {
 /// Return 32 byte random scalar, uniformly.
 #[no_mangle]
 pub extern "system" fn librustzcash_sapling_generate_r(result: *mut [c_uchar; 32]) {
-    // create random 64 byte buffer
     let mut rng = OsRng::new().expect("should be able to construct RNG");
-    let mut buffer = [0u8; 64];
-    for i in 0..buffer.len() {
-        buffer[i] = rng.gen();
-    }
+    let buffer = gen_random(&mut rng);
 
     // reduce to uniform value
-    let r = <Bls12 as JubjubEngine>::Fs::to_uniform(&buffer[..]);
+    let r = <Bls12 as JubjubEngine>::Fs::to_uniform(&buffer[0..64]);
```

Comment:
This looks wrong to me. Neither `gen_random` nor `to_uniform` (https://github.com/zcash/librustzcash/blob/e4187f07ff2dc8b9acf6ff24cc9596fbf6e4d62b/sapling-crypto/src/jubjub/fs.rs#L570-L581) do the BLAKE2b hash required by the spec.